app/controllers/adopted_controller.rb

```diff
@@ -0,0 +1,48 @@
+class AdoptedController < ApplicationController
+  before_action :authenticate
+
+  # GET /api/v1/drains/adopted
+  # Optional params:
+  #
+  #  page
+  def index
+    adopted_things
+    make_cur_page
+    make_other_pages
+    @results = {next_page: @next_page, prev_page: @prev_page, total_pages: @adopted_things.page(1).total_pages, drains: @things}
+    render json: @results
+  end
+
+private
+
+  def adopted_things
+    @adopted_things = Thing.adopted
+  end
+
+  # Determine if the user supplied a valid page number, if not they get first page
+  def make_cur_page
+    page = params[:page].blank? || params[:page].to_i.zero? ? 1 : params[:page]
+    @cur_page = @adopted_things.page(page)
```

Comment:
This coupling feels a bit weird as it requires the developer to have called `adopted_things` first.

I think this can be simplified a bit if `def index` was like:

```ruby
adopted_things = Things.adopted.order(:created_at).page(params[:page])

results = {
  next_page: adopted_things.next_page, # I think it's fine if this is null to the caller
  prev_page: adopted_things.prev_page,
  total_pages: adopted_things.total_pages,
  things: format_fields(things)
}

render json: @results
```

and then the `make_cur_pages`,  `make_other_pages` and the shared state could be removed.